dipy/reconst/multi_voxel.py

```diff
@@ -3,8 +3,140 @@
+    result: list of tuple
+        return a list of tuple(voxel index, model fitted instance)
+    """
+    model, chunks = arguments
+    return [(idx, model.fit(shared_arr[idx]))
+            for idx in chunks]
+
+
+def _init_parallel_fit_worker(arr_to_populate, shape):
+    """
+    Each pool process calls this initializer.
+    Load the array to be populated into that process's global namespace
+
+    Parameters
+    -----------
+    arr_to_populate: multiprocessing.Array
+        shared array
+    shape: tuple
+        tuple of array dimensions
+    """
+    global shared_arr
+    with warnings.catch_warnings():
+        warnings.simplefilter('ignore', RuntimeWarning)
+        shared_arr = mparray_as_ndarray(arr_to_populate, shape)
+
+
+def parallel_voxel_fit(single_voxel_fit):
+    """
+    Wraps single_voxel_fit method to turn a model into a parallel
+    multi voxel model. Use this decorator on the fit method of
+    your model to take advantage of the MultiVoxelFit.
+
+    Parameters
+    -----------
+    single_voxel_fit : callable
+        Should have a signature like: model [self when a model method is being
+        wrapped], data [single voxel data].
+
+    Returns
+    --------
+    multi_voxel_fit_function : callable
+
+    Examples
+    ---------
+    >>> import numpy as np
+    >>> from dipy.reconst.base import ReconstModel, ReconstFit
+    >>> class BasicModel(ReconstModel):
+    ...     @parallel_voxel_fit
+    ...     def fit(self, single_voxel_data):
+    ...         return ReconstFit(self, single_voxel_data.sum())
+    ...
+    >>> if __name__ == '__main__':
+    ...     data = np.random.random((2, 3, 4, 5))
+    ...     fit = model.fit(data)
+    ...     assert np.allclose(fit.data, data.sum(-1))
+    """
+
+    def new_fit(model, data, *args, **kwargs):
+        """Fit method in parallel for every voxel in data """
+        # Pop the mask, if there is one
+        mask = kwargs.get('mask', None)
+        # print(mask, mask.shape)
+        if data.ndim == 1:
+            return single_voxel_fit(model, data)
+        if mask is None:
+            mask = np.ones(data.shape[:-1], bool)
+        elif mask.shape != data.shape[:-1]:
+            raise ValueError("mask and data shape do not match")
+
+        # Get number of processes
+        nb_processes = int(kwargs.pop('nb_processes', '0'))
+        nb_processes = nb_processes if nb_processes >= 1 else cpu_count()
+
+        if nb_processes == 1:
+            return single_voxel_fit(model, data)
+
+        # Get non null indexes from mask
+        indexes = np.argwhere(mask)
+        # Convert indexes to tuple
+        indexes = [tuple(v) for v in indexes]
+        # Create chunks
+        chunks_spacing = np.linspace(0, len(indexes), num=nb_processes + 1)
+        chunks = [(indexes[int(chunks_spacing[i - 1]): int(chunks_spacing[i])])
+                  for i in range(1, len(chunks_spacing))]
+
+        # Create shared array
+        shared_arr_in = ndarray_to_mparray(data)
```

Comment:
Add in the comment that you make a single copy of the data here. If you make a copy.